### PR TITLE
Move attribute enforcement from v9 to preview

### PR DESCRIPTION
**Pull request by T-Gro**

### CODE REVIEW

**Comment by vzarytovskii** [comment]
@T-Gro did you confirm that this feature is causing the regression?

**Comment by T-Gro** [comment]
> @T-Gro did you confirm that this feature is causing the regression?

See the modified tests, the "No constructors are available for the type" issue is the same as the original. (and should not be there, because the type does have a constructor)

File: tests/fsharp/typecheck/sigs/neg110.bsl
Comment (edgarfgp) [inline tests/fsharp/typecheck/sigs/neg110.bsl:1]: I do not remember modifying this file as part of my PRs 🧐.

Comment (T-Gro) [reply tests/fsharp/typecheck/sigs/neg110.bsl:1]: The negative typecheck tests do not run with langver=preview, this was only modified when the feature was enabled by default: https://github.com/dotnet/fsharp/pull/17558/files#diff-036387972a5f139a8b0f05f5b702666c5d77d961a15f0b9ac170264d79dd1c11

**Comment by edgarfgp** [comment]
@vzarytovskii  should be this reverted now that https://github.com/dotnet/fsharp/issues/17731  has been fixed ?

**Comment by vzarytovskii** [comment]
> @vzarytovskii  should be this reverted now that https://github.com/dotnet/fsharp/issues/17731  has been fixed ? 

Yes, it won't make it to release though

**Comment by edgarfgp** [comment]
> > @vzarytovskii  should be this reverted now that #17731  has been fixed ?
> 
> Yes, it won't make it to release though

But it will be in one of the point releases right ?

**Comment by vzarytovskii** [comment]
> > > @vzarytovskii  should be this reverted now that #17731  has been fixed ?
> 
> > 
> 
> > Yes, it won't make it to release though
> 
> 
> 
> But it will be in one of the point releases right ? 

Yeah

**Pull request merged**